src/DotNetWorker.Core/Converters/JsonPocoConverter.cs

```diff
@@ -28,13 +29,11 @@ public JsonPocoConverter(IOptions<WorkerOptions> options)
             _serializer = options.Value.Serializer;
         }
 
-        public bool TryConvert(ConverterContext context, out object? target)
+        public async ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
         {
-            target = default;
-
-            if (context.Parameter.Type == typeof(string))
+            if (context.TargetType == typeof(string))
             {
-                return false;
+                return ConversionResult.Failed();
```

Comment:
Just a lack of context question: Why don't we want to convert items of type `string`? Is it just that this is specifically an object converter and if for some reason we get a string we're like "no thanks, wrong type!"

Reply:
We are checking the `TargetType`, not source type. If the TargetType is string, we already have another converter (TypeConverter) which will do that conversion. That converter is registered in the order to be used before this one.

Reply:
Ahh that makes sense, thank you :)